### PR TITLE
bpo-35018: Sax parser provides no user access to lexical handlers

### PR DESCRIPTION
**Pull request by ZackerySpytz**
Co-Authored-By: Jonathan Gossage <jgossage@gmail.com>

https://bugs.python.org/issue35018

### CODE REVIEW

File: Lib/xml/sax/handler.py [hunk 1]
```diff
@@ -340,3 +340,47 @@ def resolveEntity(self, publicId, systemId):
...
+    """Optional SAX2 handler for lexical events.
+
+    This handler is used to obtain lexical information about an XML
+    document, that is, information about how the document was encoded
+    (as opposed to what it contains, which is reported to the
+    ContentHandler), such as comments and CDATA marked section
+    boundaries.
+
+    To set the LexicalHandler of an XMLReader, use the setProperty
+    method with the property identifier."""
```
Comment (scoder) [inline Lib/xml/sax/handler.py]: I think having the property identifier string in there was needed for the documentation, even if it's not a valid URL.
```suggestion
    method with the property identifier
    'http://xml.org/sax/handlers/LexicalHandler'."""
```

**Review by scoder — COMMENTED** [review]
Looks good to me, just one issue regarding the usage documentation.

File: Lib/xml/sax/handler.py (hunk 1 above)
Comment (ZackerySpytz) [reply Lib/xml/sax/handler.py]: You're right. Thank you, @scoder, for looking at this PR.

File: Lib/xml/sax/handler.py [hunk 2]
```diff
@@ -340,3 +340,48 @@ def resolveEntity(self, publicId, systemId):
...
+
+    This handler is used to obtain lexical information about an XML
+    document, that is, information about how the document was encoded
+    (as opposed to what it contains, which is reported to the
+    ContentHandler), such as comments and CDATA marked section
+    boundaries.
+
+    To set the LexicalHandler of an XMLReader, use the setProperty
+    method with the property identifier
+    'http://xml.org/sax/handlers/LexicalHandler'."""
```
Comment (scoder) [inline Lib/xml/sax/handler.py]: I just noticed that the test actually use "http://xml.org/sax/properties/lexical-handler" instead, which is the correct property ID.
```suggestion
    'http://xml.org/sax/properties/lexical-handler'."""
```

File: Doc/library/xml.sax.handler.rst [hunk 3]
```diff
@@ -413,3 +419,45 @@ the passed-in exception object.
...
+
+This handler is used to obtain lexical information about an XML
+document. Lexical information includes information describing the
+document encoding used and XML comments embedded in the document, as
+well as section boundaries for the DTD and for any CDATA sections.
+The lexical handlers are used in the same manner as content handlers.
+
+Set the LexicalHandler of an XMLReader by using the setProperty method
+with the property identifier
+``'http://xml.org/sax/handlers/LexicalHandler'``.
```
Comment (scoder) [inline Doc/library/xml.sax.handler.rst]:
```suggestion
``'http://xml.org/sax/properties/lexical-handler'``.
```

**Comment by ZackerySpytz** [comment]
Hi, @scoder. Can this PR be merged?

**Comment by ZackerySpytz** [comment]
@scoder Ping?

**Comment by scoder** [comment]
Let's get this in. Thanks @jgossage and @ZackerySpytz for you contribution.

**Pull request merged**